batch/batch.go

```diff
@@ -154,6 +171,15 @@ func (f *Func) Invoke(ctx context.Context, arg interface{}) (interface{}, error)
 
 		// Publish the batchGroup.
 		bctx.pendingBatchGroups[fs] = bg
+	} else {
+		// This is a subsequent invocation of the batch function, so
+		// reset the intervalTimer.
+		if bg.intervalTimer.Stop() {
+			// Only reset the timer if Stop successfully
+			// stopped the timer. Otherwise, the receive channel
+			// will be populated and we'll select on it below.
```

Comment:
Could you describe the case where we fail to stop the timer? Is if we've already `select`'d below?